custom_dc/load_data.sh

```diff
@@ -0,0 +1,319 @@
+  echo_log "Completed $cmd with status:$status in $duration secs"
+}
+
+# Parse command line options
+function parse_options {
+  while (( $# > 0 )); do
+    case $1 in
+      -i) shift; INPUT_DIR=$(readlink -f "$1");;
+      -c) shift; CONFIG=$(readlink -f "$1");;
+      -k) shift; DC_API_KEY="$1";;
+      -o) shift; OUTPUT_DIR="$1";;
+      -m) shift; MODE="$1";;
+      -j) shift; DC_IMPORT_JAR="$1";;
+      -r) shift; RUN_STEPS="$1";;
+      -s) shift; USE_CLOUDSQL=true; CLOUDSQL_INSTANCE="$1";;
+      -u) shift; DB_USER="$1";;
+      -p) shift; DB_PASS="$1";;
+      -e) shift; ENV_FILE=$1; source $ENV_FILE;;
+      -q) QUIET="1";;
+      -h) echo -e "$USAGE" >&2 && exit 0;;
+      -x) set -x;;
+      *) echo "Unknown option '$1'" && exit 1;;
+    esac
+    shift
+  done
+}
+
+
+function setup_python {
+  python3 -m venv $TMP_DIR/env-stats
+  source $TMP_DIR/env-stats/bin/activate
+  if [[ "$PYTHON_REQUIREMENTS_INSTALLED" != "true" ]]
+  then
+    echo_log "Installing Python requirements from $SIMPLE_DIR/requirements.txt"
+    run_cmd pip3 install -r $SIMPLE_DIR/requirements.txt
+    PYTHON_REQUIREMENTS_INSTALLED=true
+
+    # Setup python for building NL embeddings
+    embeddings_req="$WEBSITE_DIR/tools/nl/embeddings/requirements.txt"
+    if [[ -f "$embeddings_req" ]]; then
+      echo_log "Installing Pytorch"
+      run_cmd python3 -m pip install --upgrade pip
+      run_cmd pip3 install torch==2.2.2 --extra-index-url https://download.pytorch.org/whl/cpu
+      echo_log "Installing Python requirements from $embeddings_req"
+      run_cmd pip3 install -r "$embeddings_req"
+    fi
+  fi
+}
+
+function setup_dc_import {
+  # Get the datacommons import jar
+  if [[ -f "$DC_IMPORT_JAR" ]]; then
+    echo_log "Using existing dc-import jar: $DC_IMPORT_JAR"
+  else
+    # Download the latest jar
+    echo_log "Getting latest version of dc-import jar file..."
+    # Get URLthe latest release
+    jar_url=$(curl -vs  "https://api.github.com/repos/datacommonsorg/import/releases/latest" | \
+      grep browser_download_url | cut -d\" -f4)
+    [[ -z "$jar_url" ]] && echo_fatal "Unable to get latest jar for https://github.com/datacommonsorg/import/releases.
+Please download manually and set command line option '-j'"
+    jar=$(basename $jar_url)
+    u[ -z "$DC_IMPORT_JAR" ]] && DC_IMPORT_JAR="$TMP_DIR/jar"
+    echo_log "Downloading dc-import jar from $jar_url into $DC_IMPORT_JAR..."
+    curl -Ls "$jar_url" -o $DC_IMPORT_JAR
+    [[ -f "$DC_IMPORT_JAR" ]] || echo_fatal "Failed to download $jar_url"
+  fi
+}
+
+# Setup submodules
+# import module is used for stats loading.
+function setup_submodules {
+  local cwd=$PWD
+  cd $WEBSITE_DIR
+  if [[ ! -d "import/simple" ]]; then
+    run_cmd scripts/update_git_submodules.sh
+  fi
+  if [[ -d "$WEBSITE_DIR/import/simple" ]]; then
+    SIMPLE_DIR="$WEBSITE_DIR/import/simple"
+  fi
+  cd $cwd
+}
+
+function setup {
+  WEBSITE_DIR=$(realpath $0 | sed -e 's,/website/.*,//website,')
```

Comment:
Will this work if my website repo folder is not named `website`? e.g. in my case, my website repo folder is called `dcwebsite`.

Reply:
Good catch! Changed to use dirname